### DISCUSSION TITLE
Add spacing to increase visibility for attributes in class construct

### DISCUSSION THREAD

**Discussion by Squad-G** [PHP-CS-Fixer/PHP-CS-Fixer#7111] [Feature ideas]
Say we have this method

```
public function __construct(
#[Attribute1]
#[Attribute2]
public string $hello,
#[Attribute1]
#[Attribute2]
public string $bye
){}
```

It is sometimes hard to read (we have this structure to validate DTO in Laravel).

Adding a return line between properties would greatly increase readability:

```
public function __construct(
#[Attribute1]
#[Attribute2]
public string $hello,

#[Attribute1]
#[Attribute2]
public string $bye
){}
```

Thanks 🙏

**Comment by Wirone** [comment]
@kubawerlos @keradus what do you think, should it be part of `method_argument_space` or standalone fixer? We currently don't have fixer for constructor property promotion, it would be nice place for it too. We need to decide something and make issue for it.

**Comment by kubawerlos** [comment]
`method_argument_space` sounds like a good fixer to extend with the support of such fix.

**Comment by gguenther24** [comment]
I would like this feature very much. Is there already any work on this extension of `method_argument_space`?

How would you choose to configure it? Would you just 
a) change the behavior for ['on_multiline' => 'ensure_fully_multiline', 'attribute_placement' => 'standalone'] to always have an empty line for property promotion or 
b) add a new value for the attribute 'attribute_placement' like 'standalone _with_empty_line_on_property_promotion' or
c) add a new boolean attribute like 'empty_line_on_property_promotion' => true?

Catchier name suggestions for b and c are also very welcome!

**Reply by Wirone** [reply]
Hi @gguenther24! I believe nobody has worked on it yet. IMHO it should be option C, but with `arguments_separation` (analogical to [`phpdoc_separation`](https://cs.symfony.com/doc/rules/phpdoc/phpdoc_separation.html) or [`class_attributes_separation`](https://cs.symfony.com/doc/rules/class_notation/class_attributes_separation.html) fixers) that can have value `none|one|only_if_meta`.

To consider:
- do we need `one` strategy? Do anybody would like to have multiple empty lines in method signature 🤔?
- which option should be a default? Personally I would go with `only_if_meta`, but then probably it should be configured in rulesets to use `none` to keep current behaviour.

**Comment by SchmidtClaudia** [comment]
Hi. I just want to let you know that I'm currently working on it. I try do create a pull request in the next weeks.
